### PR TITLE
Update automl.py to add verbose argument to fit

### PR DESCRIPTION
**Pull request by ImanHosseini**
as per: https://github.com/microsoft/FLAML/issues/38
Adding _verbose_ to fit: the arg is passed down to _search (instead of calling _search with verbose=0) also sets logging level to WARNING and if verbose = 0 and sets it back before returning. 

[GitHub is having issues today! It messed up my previous PR]

### CODE REVIEW

File: flaml/automl.py [hunk 1]
```diff
@@ -823,6 +824,8 @@ def custom_metric(X_test, y_test, estimator, labels,
             y_val: None | a numpy array or a pandas series of validation labels
             sample_weight_val: None | a numpy array of the sample weight of
                 validation data
+            verbose : int, default=0 | Controls the verbosity, higher means more
```
Comment (sonichi) [inline flaml/automl.py:827]: default=1?

File: flaml/automl.py [hunk 2]
```diff
@@ -906,6 +913,8 @@ def custom_metric(X_test, y_test, estimator, labels,
             self._state.n_jobs = n_jobs
             self._search()
             logger.info("fit succeeded")
+        if verbose:
```
Comment (sonichi) [inline flaml/automl.py:916]: what if verbose==0?

File: flaml/automl.py [hunk 3]
```diff
@@ -1018,7 +1027,7 @@ def _search(self):
                 init_config=None, 
                 search_alg=search_state.search_alg,
                 time_budget_s=budget_left,
-                verbose=0, local_dir='logs/tune_results',
+                verbose=self.verbose, local_dir='logs/tune_results',
```
Comment (sonichi) [inline flaml/automl.py:1030]: Could you test run `notebook/flaml_automl` to see if the output is desirable when verbose=1? If you are not sure please copy a snippet of the output from it and paste here.

**Review by sonichi — CHANGES_REQUESTED** [review]

File: flaml/automl.py (hunk 1 above)
Comment (sonichi) [reply flaml/automl.py:827]: remove the extra space before colon

File: flaml/automl.py (hunk 3 above)
Comment (ImanHosseini) [reply flaml/automl.py:1030]: I ran the notebook, if verbose is 0, what I get is:
"""
C:\Python\Python3.7\WPy64-3771\python-3.7.7.amd64\lib\site-packages\xgboost\sklearn.py:888: UserWarning: The use of label encoder in XGBClassifier is deprecated and will be removed in a future release. To remove this warning, do the following: 1) Pass option use_label_encoder=False when constructing XGBClassifier object; and 2) Encode your labels (y) as integers starting with 0, i.e. 0, 1, 2, ..., [num_class - 1]. 
"""
(multiple of these messages, because it is called on every pass, but no 'INFO' statements)
When I run with verbose = 1 I get the warnings AND I get these:
"""
[2021-03-13 14:10:53] INFO (flaml.tune.tune/MainThread) trial 1 config: {'n_estimators': 4.0, 'max_leaves': 8.0, 'min_child_weight': 1.4505165368937825, 'learning_rate': 0.12354235042058405, 'subsample': 0.9504860919941635, 'log_max_bin': 7.0, 'colsample_bytree': 0.9658058260875579, 'reg_alpha': 2.955773383057842e-10, 'reg_lambda': 0.23033924766285224, 'FLAML_sample_size': 10000}
"""
(and all other 'INFO' statements, basically verbose = 1 is as it was before this change, verbose = 0 now adds the possibility to silence those 'INFO' statements)

File: flaml/automl.py (hunk 1 above)
Comment (ImanHosseini) [reply flaml/automl.py:827]: So before adding the verbose arguments, 'INFO' statements were being printed (as if verbose is >0). but at the same time, when calling _search(...) , 0 was being passed to verbose. Now we either pick verbose default value to be 1 (so that we get those 'INFO' prints) or if we set verbose=0, then due to some previous test that checks debug buffer, we would fail that test, and then we would have to change that test.

File: flaml/automl.py (hunk 2 above)
Comment (ImanHosseini) [reply flaml/automl.py]: my bad, this should be "if verbose==0" (so that if we have previously changed logger level, we revert it to what it was before.

File: flaml/automl.py (hunk 3 above)
Comment (sonichi) [reply flaml/automl.py:1030]: My desired default behavior is: output the "INFO" in flaml; don't output the "INFO" in flaml.tune. What about the following?
default: `verbose=1`, which output "INFO" in flaml but don't output "INFO" in flaml.tune.
`verbose=0`: output no "INFO" at all.
`verbose=2`: output "INFO" in both flaml and flaml.tune.

File: flaml/automl.py (hunk 1 above)
Comment (sonichi) [reply flaml/automl.py]: Please my last comment in the bottom.

File: flaml/automl.py (hunk 3 above)
Comment (ImanHosseini) [reply flaml/automl.py:1030]: Nice! Because tune handles verbose flag already: https://github.com/microsoft/FLAML/blob/main/flaml/tune/tune.py#L227
The only thing is, tune would be at 'DEBUG' for verbose=3 and 'INFO' for verbose in {2, 1} so we only need to pass verbose-1 to tune.run and we're good.

File: flaml/automl.py [hunk 4]
```diff
@@ -823,6 +824,8 @@ def custom_metric(X_test, y_test, estimator, labels,
             y_val: None | a numpy array or a pandas series of validation labels
             sample_weight_val: None | a numpy array of the sample weight of
                 validation data
+            verbose: int, default=0 | Controls the verbosity, higher means more
```
Comment (sonichi) [inline flaml/automl.py:827]: default=1

**Review by sonichi — CHANGES_REQUESTED** [review]
looks good except the default value

File: flaml/automl.py (hunk 4 above)
Comment (ImanHosseini) [reply flaml/automl.py:827]: ouch, sorry!

**Review by sonichi — APPROVED** [review]

**Review by qingyun-wu — APPROVED** [review]
